### PR TITLE
Properly set len parameter for GifAddExtensionBlock

### LINKED ISSUE #10990: libAfterImage: TEve broken with giflib 5
- [x] Checked for duplicates (related to https://sft.its.cern.ch/jira/browse/ROOT-7904)

### Describe the bug

A patch from Gentoo imported in d9ca8cb6e3b1e4bdd70041c43805f0851aea7c9e (https://gitweb.gentoo.org/repo/gentoo.git/plain/media-libs/libafterimage/files/libafterimage-giflib5-v2.patch) contains a bogus call
```diff
+#if (GIFLIB_MAJOR>=5)
+            		if ((status = GifAddExtensionBlock(&temp_save.ExtensionBlockCount, &temp_save.ExtensionBlocks,
+                            ExtCode, sizeof(ExtData), ExtData)) == GIF_OK)
+                    status = DGifGetExtension(gif,&ExtCode,&ExtData);
+#else
```
where
```
     GifByteType *ExtData;
```
so `sizeof(ExtData)` is a size of a pointer when it's probably has to be a size of some data block.

Which results in a crash when trying to use TEve:
```
root [0] TEveManager::Create()
gif2ASImage():2300:</nix/store/jhzzgdncnmwzdb2jp9dhmv9bb2nyxcl3-root-6.24.06/icons/eve_text.gif> (null)

 *** Break *** segmentation violation
```

### Expected behavior

TEve works.

### Setup

ROOT 6.24.06 compiled against an external libAfterImage, with the same Gentoo patches applied to compile against giflib 5 
Crash reproduced on macOS.

### PR DESCRIPTION
**Pull request by linev**
May be fixes #10990

### CODE REVIEW

File: graf2d/asimage/src/libAfterImage/ungif.c [hunk 1]
```diff
@@ -172,7 +172,7 @@ get_gif_image_desc( GifFileType *gif, SavedImage *im )
 #else
 			im->ImageDesc.ColorMap = MakeMapObject(gif->Image.ColorMap->ColorCount, NULL);
 #endif
-			fseek( gif->UserData, start_pos+9, SEEK_SET ); 
+			fseek( gif->UserData, start_pos+9, SEEK_SET );
```
Comment (veprbl) [inline graf2d/asimage/src/libAfterImage/ungif.c:175]: May I ask you to revert this hunk? I'd like to be able to fetch this patch and apply it to the upstream tarball.

File: graf2d/asimage/src/libAfterImage/ungif.c [hunk 2]
```diff
@@ -229,8 +230,9 @@ get_gif_saved_images( GifFileType *gif, int subimage, SavedImage **ret, int *ret
 				{
             		/* Create an extension block with our data */
 #if (GIFLIB_MAJOR>=5)
+				      Len = EGifGCBToExtension(gif, ExtData);
```
Comment (veprbl) [inline graf2d/asimage/src/libAfterImage/ungif.c:233]: By coincidence this returns `4` in runtime, which should fit in `sizeof(void*)` on most platforms.

File: graf2d/asimage/src/libAfterImage/ungif.c [hunk 3]
```diff
@@ -229,8 +230,9 @@ get_gif_saved_images( GifFileType *gif, int subimage, SavedImage **ret, int *ret
...
             		if ((status = GifAddExtensionBlock(&temp_save.ExtensionBlockCount, &temp_save.ExtensionBlocks,
-                            ExtCode, sizeof(ExtData), ExtData)) == GIF_OK)
+                            ExtCode, Len, ExtData)) == GIF_OK)
                     status = DGifGetExtension(gif,&ExtCode,&ExtData);
```
Comment (veprbl) [inline graf2d/asimage/src/libAfterImage/ungif.c:236]:
```suggestion
                    status = DGifGetExtensionNext(gif, &ExtData);
```

**Review by veprbl — COMMENTED** [review]
This doesn't work, but I've managed to guess another change that fixes the issue for me.

File: graf2d/asimage/src/libAfterImage/ungif.c (hunk 3 above)
Comment (linev) [reply graf2d/asimage/src/libAfterImage/ungif.c:236]: This is the only change you need?

Comment (linev) [reply graf2d/asimage/src/libAfterImage/ungif.c:236]: Please propose your own PR

**Comment by linev** [comment]
I close PR, please create new one which works for you

**Pull request closed without merging**